Jenkinsfile

```diff
@@ -0,0 +1,75 @@
+#! groovy
+library 'pipeline-library'
+// TODO: Could we make this an array and test across multiple major versions
+def nodeVersion = '8.9.1'
+
+def unitTests(os, nodeVersion) {
+  return {
+    node(os) {
+      nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+        stage('Test') {
+          timeout(15) {
+            unstash 'sources'
+            // Install yarn if not installed
+            if('windows'.equals(os)) {
+              if (bat(returnStatus: true, script: 'where yarn') != 0) {
+                bat 'npm install -g yarn'
+              }
+              bat 'yarn install'
+            } else {
+              if (sh(returnStatus: true, script: 'which yarn') != 0) {
+                sh 'npm install -g yarn'
+              }
+              sh 'yarn install'
+           }
+           fingerprint 'package.json'
+            try {
+              if('windows'.equals(os)) {
+                bat 'yarn run coverage'
+              } else {
+                sh 'yarn run coverage'
+              }
+            } finally {
+              // record results even if tests/coverage 'fails'
+              junit 'junit.xml'
```

Comment:
You may want to "hack" the generated test report to differentiate the tests for each OS. It won't be easy to determine which OS failed a given test when they're all merged with the same names. For the SDK I inject the platform name as a prefix to the test names.

Not sure if there's a better/more official way of handling running the same tests on multiple platforms and merging them together via JUnit/Jenkins.